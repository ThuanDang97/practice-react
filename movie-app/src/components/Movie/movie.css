.movie {
  width: 100%;
  max-height: 600px;
}

.banner {
  max-width: 1440px;
  display: flex;
  margin: 0 auto;
}

.banner .banner_image {
  position: absolute;
  right: 0;
  height: 600px;
  width: 1520px;
}

.banner .layer {
  width: 100%;
  height: 600px;
  position: absolute;
  right: 0;
  background-image: linear-gradient(
    to right,
    rgba(0, 0, 0, 1) 150px,
    rgba(220.5, 220.5, 220.5, 0.84) 100%
  );
}

.banner .content {
  z-index: 1;
  padding: 75px 40px;
}
