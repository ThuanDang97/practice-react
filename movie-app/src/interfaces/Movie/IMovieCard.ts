import { IMovie } from './IMovie';

export interface IMovieCard extends IMovie {
  is_favorite: boolean;
}
