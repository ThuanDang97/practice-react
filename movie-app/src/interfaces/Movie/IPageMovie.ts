import { IMovie } from './IMovie';

export interface IPageMovie {
  page: number;
  results: IMovie[];
  total_pages: number;
  total_results: number;
}
