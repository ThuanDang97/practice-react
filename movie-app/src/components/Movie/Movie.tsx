import { IMovieInfo } from '../../interfaces/Movie/IMovieInfo';
import ContentMovie from './ContentMovie/ContentMovie';
import './movie.css';

type TProps = {
  movie: IMovieInfo;
};

const Movie = (props: TProps): JSX.Element => {
  const { movie } = props;

  return (
    <section className="movie">
      <div className="banner">
        <img
          src={`https://www.themoviedb.org/t/p/w1920_and_h800_multi_faces${movie?.backdrop_path}`}
          alt={movie?.original_title}
          className="banner_image"
        />
        <span className="layer"></span>
        <ContentMovie
          posterPath={movie?.poster_path!}
          homepage={movie?.homepage!}
          title={movie?.title!}
          releaseDate={movie?.release_date!}
          genres={movie?.genres!}
          runtime={movie?.runtime!}
          overview={movie?.overview!}
          scoreVoteAverage={movie?.vote_average!}
        />
      </div>
    </section>
  );
};

export default Movie;
