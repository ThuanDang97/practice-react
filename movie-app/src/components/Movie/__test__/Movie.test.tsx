import { screen, render } from '@testing-library/react';
import { IMovieInfo } from '../../../interfaces/Movie/IMovieInfo';
import Movie from '../Movie';
import { TProps } from '../Movie';

describe('Test Movie component', () => {
  const mockData: IMovieInfo = {
    adult: false,
    backdrop_path: '/mBJ995JUssgh2hhm4JxNj8hyVVM.jpg',
    belongs_to_collection: [],
    budget: 0,
    genres: [],
    homepage:
      'https://www.disneyplus.com/movies/beyond-infinity-buzz-and-the-journey-to-lightyear/6UHRsekPfYyP',
    id: 979163,
    imdb_id: 'tt16026684',
    original_language: 'en',
    original_title: 'Beyond Infinity: Buzz and the Journey to Lightyear',
    overview:
      'Explore the evolution of Buzz Lightyear from toy to human in the making of Pixar’s Lightyear. Dive into the origin and cultural impact of everyone’s favorite Space Ranger, the art of designing a new “human Buzz,” and the challenges faced by the Lightyear crew along the way.',
    popularity: 1852.057,
    poster_path: '/3dhQksZrIeRYP8jbB0J1SLW6HVW.jpg',
    production_companies: [],
    production_countries: [],
    release_date: '2022-06-10',
    revenue: 0,
    runtime: 35,
    spoken_languages: [],
    status: 'Released',
    tagline: '',
    title: 'Beyond Infinity: Buzz and the Journey to Lightyear',
    video: false,
    vote_average: 7.4,
    vote_count: 100,
    is_favorite: false,
    genre_ids: [],
  };

  const defaultProps: TProps = {
    movie: mockData,
  };

  const component = <Movie {...defaultProps} />;

  test('Component Movie matches DOM Snapshot', () => {
    const { container } = render(component);

    expect(container).toMatchSnapshot();
  });

  test('Component Movie should render correctly', () => {
    render(component);

    // banner should be render and match alt
    expect(screen.getByTestId('banner')).toBeInTheDocument();
    expect(screen.getAllByRole('img')[0]).toBeTruthy();

    // poster should be render and match alt
    expect(screen.getByTestId('poster')).toBeInTheDocument();
    expect(screen.getAllByRole('img')[1]).toBeTruthy();
    expect(screen.getByTestId('title')).toHaveTextContent(
      'Beyond Infinity: Buzz and the Journey to Lightyear',
    );
    expect(screen.getByTestId('release')).toHaveTextContent('2022');
    expect(screen.getByTestId('runtime')).toHaveTextContent('0h 35m');
    expect(screen.getByTestId('score')).toHaveTextContent('74');
    expect(screen.getByTestId('desc')).toHaveTextContent(
      'Explore the evolution of Buzz Lightyear from toy to human in the making of Pixar’s Lightyear. Dive into the origin and cultural impact of everyone’s favorite Space Ranger, the art of designing a new “human Buzz,” and the challenges faced by the Lightyear crew along the way.',
    );
  });
});
