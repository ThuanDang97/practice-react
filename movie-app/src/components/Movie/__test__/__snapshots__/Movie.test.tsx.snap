// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Movie component Component Movie matches DOM Snapshot 1`] = `
<div>
  <section
    class="movie"
  >
    <div
      class="banner"
    >
      <img
        alt="Beyond Infinity: Buzz and the Journey to Lightyear"
        class="banner_image"
        data-testid="banner"
        src="https://www.themoviedb.org/t/p/w1920_and_h800_multi_faces/mBJ995JUssgh2hhm4JxNj8hyVVM.jpg"
      />
      <span
        class="layer"
      />
      <div
        class="content"
      >
        <div
          class="inner"
        >
          <div
            class="poster_movie"
          >
            <img
              alt="https://www.disneyplus.com/movies/beyond-infinity-buzz-and-the-journey-to-lightyear/6UHRsekPfYyP"
              class="image_movie"
              data-testid="poster"
              src="https://www.themoviedb.org/t/p/w300_and_h450_bestv2/3dhQksZrIeRYP8jbB0J1SLW6HVW.jpg"
            />
          </div>
          <section
            class="info_movie"
          >
            <div
              class="info_wrapper"
            >
              <div
                class="tittle"
              >
                <h2>
                  <a
                    data-testid="title"
                    href="#"
                  >
                    Beyond Infinity: Buzz and the Journey to Lightyear
                  </a>
                  <span
                    class="tag release_date"
                    data-testid="release"
                  >
                    (
                    2
                    0
                    2
                    2
                    )
                  </span>
                </h2>
                <div
                  class="facts"
                >
                  <span
                    class="genres"
                    data-testid="genres"
                  />
                  <span
                    class="runtime"
                    data-testid="runtime"
                  >
                    0h 35m
                  </span>
                </div>
              </div>
              <ul
                class="options"
              >
                <li
                  class="chart"
                >
                  <div
                    class="details"
                  >
                    <div
                      class="outer_ring"
                    >
                      <div
                        class="user_score_chart"
                      >
                        <div
                          class="percent"
                        >
                          <span
                            class="icon icon-r73"
                            data-testid="score"
                          >
                            74
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="text"
                  >
                    User
                    <br />
                    Score
                  </div>
                </li>
                <div
                  class="tooltip"
                  data-testid="tooltip"
                >
                  <div>
                    <li
                      class="option-single"
                    >
                      <i>
                        <img
                          alt="Add to list"
                          src="https://www.themoviedb.org/assets/2/v4/glyphicons/basic/glyphicons-basic-159-thumbnails-list-white-c260ea972eebf812289fd3c41d0d2c1dff33ecbcd62be13fba8eeaf9de173789.svg"
                        />
                      </i>
                    </li>
                  </div>
                  <span
                    class="tooltip-text"
                  >
                    Add to list
                  </span>
                </div>
                <div
                  class="tooltip"
                  data-testid="tooltip"
                >
                  <div>
                    <li
                      class="option-single"
                    >
                      <i>
                        <img
                          alt="Favorite"
                          src="https://www.themoviedb.org/assets/2/v4/glyphicons/basic/glyphicons-basic-13-heart-white-28d2cc2d6418c5047efcfd2438bfc5d109192671263c270993c05f130cc4584e.svg"
                        />
                      </i>
                    </li>
                  </div>
                  <span
                    class="tooltip-text"
                  >
                    Remove from your favorite list
                  </span>
                </div>
                <div
                  class="tooltip"
                  data-testid="tooltip"
                >
                  <div>
                    <li
                      class="option-single"
                    >
                      <i>
                        <img
                          alt="Watchlist"
                          src="https://www.themoviedb.org/assets/2/v4/glyphicons/basic/glyphicons-basic-73-bookmark-white-432e98d550b7e4c80b06272c49475b0db85a60f6fae450420713004b3c9d3ffd.svg"
                        />
                      </i>
                    </li>
                  </div>
                  <span
                    class="tooltip-text"
                  >
                    Add to your watchlist
                  </span>
                </div>
                <div
                  class="tooltip"
                  data-testid="tooltip"
                >
                  <div>
                    <li
                      class="option-single"
                    >
                      <i>
                        <img
                          alt="Your rating"
                          src="https://www.themoviedb.org/assets/2/v4/glyphicons/basic/glyphicons-basic-49-star-white-5c85220678b312aea9599d5f12ad858a9e7df226de51ef8b6b699023ffeda5fa.svg"
                        />
                      </i>
                    </li>
                  </div>
                  <span
                    class="tooltip-text"
                  >
                    Rate It!
                  </span>
                </div>
              </ul>
              <div
                class="overview"
              >
                <h3>
                  Overview
                </h3>
                <p
                  class="desc"
                  data-testid="desc"
                >
                  Explore the evolution of Buzz Lightyear from toy to human in the making of Pixar’s Lightyear. Dive into the origin and cultural impact of everyone’s favorite Space Ranger, the art of designing a new “human Buzz,” and the challenges faced by the Lightyear crew along the way.
                </p>
              </div>
            </div>
          </section>
        </div>
      </div>
    </div>
  </section>
</div>
`;
